ISM43362Interface.cpp

```diff
@@ -443,14 +444,20 @@ int ISM43362Interface::socket_send_nolock(void *handle, const void *data, unsign
 int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
 {
     _mutex.lock();
+    int ret = socket_recv_nolock(handle, data, size);
+    _mutex.unlock();
+    return ret;
+}
+
+int ISM43362Interface::socket_recv_nolock(void *handle, void *data, unsigned size)
+{
     unsigned recv = 0;
     struct ISM43362_socket *socket = (struct ISM43362_socket *)handle;
     char *ptr = (char *)data;
 
     // debug_if(_ism_debug, "ISM43362Interface socket_recv: req=%d read_data_size=%d connected %d\r\n", size, socket->read_data_size, socket->connected);
 
     if (!socket->connected) {
-        _mutex.unlock();
```

Comment:
You probably can't remove this? The lock is acquired in the beginning of this function.